api/src/main/java/org/apache/gravitino/model/Model.java

```diff
@@ -0,0 +1,180 @@
+ * under the License.
+ */
+package org.apache.gravitino.model;
+
+import java.util.Collections;
+import java.util.Map;
+import org.apache.gravitino.Auditable;
+import org.apache.gravitino.Namespace;
+import org.apache.gravitino.annotation.Evolving;
+import org.apache.gravitino.authorization.SupportsRoles;
+import org.apache.gravitino.exceptions.ModelVersionAliasesAlreadyExistsException;
+import org.apache.gravitino.exceptions.NoSuchModelException;
+import org.apache.gravitino.exceptions.NoSuchModelVersionException;
+import org.apache.gravitino.tag.SupportsTags;
+
+/**
+ * An interface representing an ML model under a schema {@link Namespace}. A model is a metadata
+ * object that represents the model artifact in ML. Users can register a model object in Gravitino
+ * to manage the mode metadata. The typical use case is to manage the model in ML lifecycle with a
+ * unified way in Gravitino, and access the model artifact with a unified identifier. Also, with the
+ * model registered in Gravitino, users can also govern the model with Gravitino's unified audit,
+ * tag, and role management.
+ *
+ * <p>The difference of Model and tabular data is that the model is schema-free, and the main
+ * property of the model is the model artifact URL. The difference compared to the fileset is that
+ * the model is versioned, and the model object contains the version information.
+ */
+@Evolving
+public interface Model extends Auditable {
+
+  /** @return Name of the model object. */
+  String name();
+
+  /**
+   * The comment of the model object. This is the general description of the model object. User can
+   * still add more detailed information in the model version.
+   *
+   * @return The comment of the model object. Null is returned if no comment is set.
+   */
+  default String comment() {
+    return null;
+  }
+
+  /**
+   * The properties of the model object. The properties are key-value pairs that can be used to
+   * store additional information of the model object. The properties are optional.
+   *
+   * <p>Users can still specify the properties in the model version for different information.
+   *
+   * @return the properties of the model object.
+   */
+  default Map<String, String> properties() {
+    return Collections.emptyMap();
+  }
+
+  /**
+   * The versions of the model object. The model object can have multiple versions. Each version
+   * contains the detailed information of the model artifact, like the URL, the training properties,
+   * etc.
+   *
+   * @return The versions of the model object.
+   */
+  ModelVersion[] versions();
+
+  /**
+   * Get the model version by the version number. If the version does not exist, throw an exception.
+   *
+   * @param version The version number of the model.
+   * @return The model version object.
+   * @throws NoSuchModelVersionException if the version does not exist.
+   */
+  ModelVersion version(int version) throws NoSuchModelVersionException;
```

Comment:
Is it more scalable to make Model Version related operations independent of interfaces?

Reply:
I cannot get your meaning, can you explain more about the meaning of "scalable"?

Reply:
I personally think that the frequency of model version metadata operations may be more frequent than some metadata of the model itself, so perhaps the model version operations can be made into a separate interface to avoid frequent modifications to the model interface implementation.

Reply:
How about also moving these interfaces to `ModelCatalog`?

In my design doc, I put these interfaces in `ModelCatalog`, so users can directly operate the `ModelVersion`, no need to get the `Model` first, and then get the `ModelVersion`. But I changed to this to align with table/partition design and concept.

Reply:
Yeah, I think these methods are probably more suitable for the definition of metadata operations rather than metadata definitions.